atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/internalschema/TransactionSchemaManager.java

```diff
@@ -78,16 +78,52 @@ public int getTransactionsSchemaVersion(long timestamp) {
      * bound, the transactions schema version is equal to newVersion.
      */
     public boolean tryInstallNewTransactionsSchemaVersion(int newVersion) {
-        List<Integer> presentVersionPeakValidity = coordinationService.tryTransformCurrentValue(
-                valueAndBound -> installNewVersionInMapOrDefault(newVersion, valueAndBound))
-                .existingValues()
-                .stream()
-                .map(valueAndBound -> valueAndBound.value()
-                        .orElseThrow(() -> new SafeIllegalStateException("Unexpectedly found no value in store"))
-                        .timestampToTransactionsTableSchemaVersion()
-                        .getValueForTimestamp(valueAndBound.bound()))
-                .collect(Collectors.toList());
-        return Iterables.getOnlyElement(presentVersionPeakValidity) == newVersion;
+        CheckAndSetResult<ValueAndBound<InternalSchemaMetadata>> transformResult = tryInstallNewVersion(newVersion);
+
+        Map.Entry<Range<Long>, Integer> finalVersion = getRangeAtBoundThreshold(
+                Iterables.getOnlyElement(transformResult.existingValues()));
```

Comment:
Just an observation that it's strange to return a CAS result that has a list of existing values, when we know it is always going to be a single value. Probably OK, but maybe it's worth having a special superclass of CheckAndSetResult<ValueAndBound<T>> (or even not generic) and have a method on it that returns the single existing value.

Reply:
Yep, that makes sense. I've checked and neither CASR nor VAB are serialized, so changing this in the future at least right now probably won't be too bad.